File: app/src/main/java/com/example/biblio/viewmodels/PopularViewModel.java
package com.example.biblio.viewmodels;

import java.util.List;

import lrusso96.simplebiblio.core.Ebook;
import lrusso96.simplebiblio.core.SimpleBiblio;

public class PopularViewModel extends SwipeEbooksViewModel {

    @Override
    protected List<Ebook> doRefresh(SimpleBiblio sb) {
        return sb.getAllPopular();
    }
}

File: app/src/main/java/com/example/biblio/viewmodels/PopularViewModel.kt
package com.example.biblio.viewmodels

import android.app.Application
import lrusso96.simplebiblio.core.Ebook
import lrusso96.simplebiblio.core.SimpleBiblio

class PopularViewModel(application: Application) : SwipeEbooksViewModel(application) {
    override suspend fun doRefresh(sb: SimpleBiblio): List<Ebook> = sb.allPopular
}